src/dask_awkward/lib/io/io.py

```diff
@@ -639,6 +687,11 @@ def from_map(
     else:
         result = new_array_object(hlg, name, meta=array_meta, npartitions=len(inputs))
 
+    if empty_on_raise and empty_backend:
+        res = result.map_partitions(first, meta=array_meta, output_divisions=1)
+        rep = result.map_partitions(second, meta=empty_typetracer())
```

Comment:
Here `res` is the actual interesting `Array` collection. The `array_meta` object is the meta we'd normally expect without going through the process of dual returns with the report. We can manually pass that meta object to `map_partitions` to override its default behavior of automatically determining a new meta, we also make sure to preserve divisions if they're known.

And `rep` is the new report `Array` collection, it doesn't matter what the meta or divisions are, this should be a pretty small object once computed. If/when we converge on a proper schema for a report array perhaps we can pass in its typetracer version for absolute correctness here

Reply:
Can we not specify `meta` and let dask-awkward figure it out using the graph?